src/libstd/process.rs

```diff
@@ -0,0 +1,835 @@
+    }
+
+    /// Executes a command as a child process, waiting for it to finish and
+    /// collecting its exit status.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::old_io::Command;
+    ///
+    /// let status = match Command::new("ls").status() {
+    ///     Ok(status) => status,
+    ///     Err(e) => panic!("failed to execute process: {}", e),
+    /// };
+    ///
+    /// println!("process exited with: {}", status);
+    /// ```
+    pub fn status(&self) -> IoResult<ExitStatus> {
+        self.spawn().and_then(|mut p| p.wait())
+    }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl fmt::Debug for Command {
+    /// Format the program and arguments of a Command for display. Any
+    /// non-utf8 data is lossily converted using the utf8 replacement
+    /// character.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{:?}", self.inner.program));
+        for arg in &self.inner.args {
+            try!(write!(f, " '{:?}'", arg));
+        }
+        Ok(())
+    }
+}
+
+impl AsInner<CommandImp> for Command {
+    fn as_inner(&self) -> &CommandImp { &self.inner }
+}
+
+impl AsInnerMut<CommandImp> for Command {
+    fn as_inner_mut(&mut self) -> &mut CommandImp { &mut self.inner }
+}
+
+fn setup_io(io: &Stdio, fd: libc::c_int, readable: bool)
+            -> IoResult<(Option<FileDesc>, Option<FileDesc>)>
+{
+    use self::Stdio::*;
+    Ok(match *io {
+        Null => {
+            (None, None)
+        }
+        Inherit => {
+            (Some(FileDesc::new(fd, false)), None)
+        }
+        Capture => {
+            let (reader, writer) = try!(unsafe { sys::os::pipe_shim() });
+            if readable {
+                (Some(reader), Some(writer))
+            } else {
+                (Some(writer), Some(reader))
+            }
+        }
+    })
+}
+
+/// The output of a finished process.
+#[derive(PartialEq, Eq, Clone)]
+pub struct Output {
+    /// The status (exit code) of the process.
+    pub status: ExitStatus,
+    /// The data that the process wrote to stdout.
+    pub output: Vec<u8>,
+    /// The data that the process wrote to stderr.
+    pub error: Vec<u8>,
+}
+
+/// Describes what to do with a standard io stream for a child process.
+// FIXME: should be an extensible enum
+#[derive(Clone)]
+pub enum Stdio {
+    /// A new pipe should be arranged to connect the parent and child processes.
+    Capture,
+
+    /// The child inherits from the corresponding parent descriptor.
+    Inherit,
+
+    /// This stream will be ignored. This is the equivalent of attaching the
+    /// stream to `/dev/null`
+    Null,
+}
+
+/// Describes the result of a process after it has terminated.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum ExitStatus {
+    /// Normal termination with an exit code.
+    Code(i32),
+
+    /// Termination by signal, with the signal number.
+    ///
+    /// Never generated on Windows.
+    Signal(i32),
+}
```

Comment:
No, I changed my mind on this. In particular, the API the RFC suggested was almost identical to this, except that rather than two variants you just got an `Option` of the first variant. Given that, I don't think there's a good reason to hide these details.